aiida/engine/daemon/client.py

```diff
@@ -95,6 +102,14 @@ def daemon_name(self) -> str:
         """
         return self._DAEMON_NAME.format(name=self.profile.name)
 
+    @contextmanager
+    def lock(self):
+        if self.is_daemon_running:
+            raise RuntimeError(f"Cannot lock the '{self.profile.name}' daemon whilst it is running.")
+        Path(self.daemon_lock_file).touch()
```

Comment:
I think here you need to avoid concurrent creation of the file to succeed silently, so `exist_ok=False`